### PR TITLE
[Renew] custom rules support sonar 9.9

### PR DESCRIPTION
**Pull request by smileQiny**

### CODE REVIEW

**Comment by smileQiny** [comment]
test demo：image
image

File: pom.xml [hunk 1]
```diff
@@ -10,21 +10,20 @@
...
-      <version>${sonar.version}</version>
+      <version>7.9</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
       <artifactId>sonar-cxx-plugin</artifactId>
       <type>sonar-plugin</type>
-      <version>0.9.7-SNAPSHOT</version>
+      <version>2.1.1-SNAPSHOT</version>
```
Comment (guwirth) [inline pom.xml:26]: Think this must be the version of the plugin. The "-SNAPSHOT" is replaced with the build number.

File: pom.xml [hunk 2]
```diff
@@ -60,8 +59,8 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.8.1</version>
         <configuration>
-          <source>1.7</source>
-          <target>1.7</target>
+          <source>10</source>
```
Comment (guwirth) [inline pom.xml:62]: This is 

        <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-compiler-plugin</artifactId>
          <version>${maven-compiler.version}</version>
          <configuration>
            <encoding>UTF-8</encoding>
            <showDeprecation>true</showDeprecation>
            <release>${java.version}</release>
            <debug>true</debug>
          </configuration>
        </plugin>

with 
<java.version>11</java.version>
<maven-compiler.version>3.12.1</maven-compiler.version>

see https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/pom.xml

File: pom.xml [hunk 3]
```diff
@@ -10,21 +10,20 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <sonar.version>8.6.1.40680</sonar.version>
   </properties>
 
   <dependencies>
     <dependency>
       <groupId>org.sonarsource.sonarqube</groupId>
       <artifactId>sonar-plugin-api</artifactId>
```
Comment (guwirth) [inline pom.xml:18]: Would keep this:

      <dependency>
        <groupId>org.sonarsource.api.plugin</groupId>
        <artifactId>sonar-plugin-api</artifactId>
        <version>${sonar.plugin.api.version}</version>
      </dependency>
      <dependency>
        <groupId>org.sonarsource.sonarqube</groupId>
        <artifactId>sonar-plugin-api-impl</artifactId>
        <version>${sonar.version}</version>
      </dependency>
      <dependency>
        <groupId>org.sonarsource.sonarqube</groupId>
        <artifactId>sonar-testing-harness</artifactId>
        <version>${sonar.version}</version>
      </dependency>

with
    <sonar.version>9.9.0.65466</sonar.version>
    <sonar.plugin.api.version>9.14.0.375</sonar.plugin.api.version>

**Review by guwirth — CHANGES_REQUESTED** [review]
HI @smileQiny,

looking to the code and past issues:
* https://github.com/SonarOpenCommunity/sonar-cxx/pull/667
* https://github.com/SonarOpenCommunity/sonar-cxx/issues/1308

A solution must use:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/d9a044196e2679301a180761dbae3a424ebc2e95/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/CustomCxxRulesDefinition.java#L33

The CxxSquidSensor is using CustomCxxRulesDefinition:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/d9a044196e2679301a180761dbae3a424ebc2e95/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/CxxSquidSensor.java#L116 

Finally addCustomChecks is called:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/d9a044196e2679301a180761dbae3a424ebc2e95/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/CxxChecks.java#L57

Maybe I understand it wrong? But current solution seems to be different?

Regards,

**Comment by guwirth** [comment]
@smileQiny please have a look to #24

**Comment by guwirth** [comment]
I merge this one and later one #24

**Pull request merged**